models/articles.models.js

```diff
@@ -14,9 +14,30 @@ exports.selectArticleById = (article_id) => {
     });
 };
 
-exports.selectArticles = () => {
+exports.selectArticles = (sort_by, order) => {
   let queryStr =
-    "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url ORDER BY created_at DESC;";
+    "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url";
+
+  const validColumns = ["author", "title", "article_id", "topic", "created_at", "votes", "article_img_url"]
+
+  if(sort_by){
+    if(!validColumns.includes(sort_by, order)){
+      return Promise.reject({ status: 400, message : 'Bad request'})
+    } else {
+      queryStr += ` ORDER BY ${sort_by};`
+    }
+  } else {
+    queryStr += ` ORDER BY created_at;`
+  }
+
+  if(order){
+    order = order.toUpperCase()
+    queryStr=queryStr.slice(0, -1)
+    queryStr+= ` ${order};`
+  } else {
+    queryStr=queryStr.slice(0, -1)
+    queryStr+= ` DESC;`
+  }
```

Comment:
Worth thinking about the validity of this order variable. If you're not validating it, then a malicious actor could SQL inject your database by passing in `ASC; DROP TABLE articles;` for example. To protect against this we'd want to take similar measures to what you've done for sort_by